Configure eslint for mocha globals

bergarces opened the pull request:
- Added `.eslintrc.json` files to stop commit hooks from wrongly flagging errors.
- Fixed some of the flagged errors in test files.
- Not all errors from all packages have been fixed. Truffle test files that make use of global variables such as `artifacts` are still affected.
- An alternative to adding a `.eslintrc.json` file to every root folder, would be to add the filepath to the mocha exception list in the root `.eslintrc.json`.

gnidan commented on packages/box/.eslintrc.json:
```diff
@@ -0,0 +1,10 @@
+      }
+    }
+  ]
+}
```
New line at end of file :)

gnidan commented on packages/compile-solidity/.eslintrc.json:
```diff
@@ -0,0 +1,10 @@
+      }
+    }
+  ]
+}
```
New line at end of file :)

gnidan commented on packages/compile-vyper/.eslintrc.json:
```diff
@@ -0,0 +1,10 @@
+      }
+    }
+  ]
+}
```
New line at end of file (don't break my `cat` please!)

gnidan commented on packages/contract-schema/.eslintrc.json:
```diff
@@ -0,0 +1,10 @@
+      }
+    }
+  ]
+}
```
new line at end of file 🦜

gnidan commented on packages/contract-tests/.eslintrc.json:
```diff
@@ -0,0 +1,10 @@
+      }
+    }
+  ]
+}
```
y'know maybe just search and replace them all :)

gnidan reviewed (changes requested):
This looks great, apart from the comment you'll see a lot of :)

Thanks for doing this! It'll be nice to have working editor integration

bergarces commented:
> This looks great, apart from the comment you'll see a lot of :)
> 
> Thanks for doing this! It'll be nice to have working editor integration

Added newlines and changed my editor settings.

gnidan reviewed (approved):
Looks good!
